### PR TITLE
[SYCL-MLIR][Test E2E] Make tests that time out fail early

### PR DESCRIPTION
**Pull request by victor-eds**
Add error directive to test.

### CODE REVIEW

File: sycl/test-e2e/SubGroup/scan_spirv13.cpp [hunk 1]
```diff
@@ -14,6 +14,8 @@
 // This test verifies the correct work of SPIR-V 1.3 exclusive_scan() and
 // inclusive_scan() algoriths used with the operation MUL, bitwise OR, XOR, AND.
 
+#error FIXME: Test times-out
```
Comment (whitneywhtsang) [inline sycl/test-e2e/SubGroup/scan_spirv13.cpp:17]: Another way to fix this is to change XFAIL on line 9 to UNSUPPORTED, which way do you think is better?

Comment (whitneywhtsang) [reply sycl/test-e2e/SubGroup/scan_spirv13.cpp:17]: `GroupAlgorithm/SYCL2020/reduce_over_group_size.cpp` too.

Comment (victor-eds) [reply sycl/test-e2e/SubGroup/scan_spirv13.cpp:17]: > `GroupAlgorithm/SYCL2020/reduce_over_group_size.cpp` too.

I had to follow a different a approach for that one, but it's `XFAIL`ing now

**Review by whitneywhtsang — APPROVED** [review]

File: sycl/test-e2e/SubGroup/scan_spirv13.cpp (hunk 1 above)
Comment (whitneywhtsang) [reply sycl/test-e2e/SubGroup/scan_spirv13.cpp:17]: You may not need to follow a different approach for that one, as it is marked XFAIL in the xfail_test.txt. And XFAIL test cases are not run on GPU.

Comment (victor-eds) [reply sycl/test-e2e/SubGroup/scan_spirv13.cpp:17]: Let's try

**Review by etiotto — APPROVED** [review]

File: sycl/test-e2e/SubGroup/scan_spirv13.cpp (hunk 1 above)
Comment (whitneywhtsang) [reply sycl/test-e2e/SubGroup/scan_spirv13.cpp:17]: Looks like the `#error` way doesn't solve the time out issue:
```
Timed Out Tests (3):
  SYCL :: SubGroup/scan_spirv13.cpp
  SYCL :: USM/copy2d.cpp
  SYCL :: USM/memcpy2d.cpp
```

Comment (victor-eds) [reply sycl/test-e2e/SubGroup/scan_spirv13.cpp:17]: `cgeist` has incorrect behavior here. https://github.com/intel/llvm/pull/9306 fixes that and should be merged before this

**Pull request merged**